spec/problem2_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'problem2'
+describe 'problem2' do
+  it 'returns the two largest values from array' do
+    expect(problem2([1,2,3,4,5])).to match_array [4,5]
+    expect(problem2([5,4,3,2,1])).to match_array [4,5]
+    expect(problem2([1,5,2,4,3])).to match_array [4,5]
+    expect(problem2([1,4,2,4,3])).to match_array [4,4]
+    expect(problem2([1,1])).to       match_array [1,1]
+  end
+
+  it 'raises ArgumentError when there are\'t two large values' do
+    expect { problem2([1]) }.to raise_error(ArgumentError)
```

Comment:
Interesting approach, I guess is more of a preference,
I believe that code should flow naturally and exceptions should be used in extreme situations. 
So in this case when the array has less than 2 elements I would prefer to return an empty array. one reason being that if some other code will use the result it will make for a better experience than handling an exception.

Can you give some arguments on why raising an exception is also a good way to go?

Reply:
Given the problem definition, I would consider the function "constrained" to return two values. If it can't it doesn't confine to the expectation of what it's supposed to do, and then it would throw an exception. Handling an exception is generally more cumbersome, I agree, but then the value checks are burdened on the callers of this function and it can become easier (theoretically) to have nil exceptions pop up somewhere else, farther from the source of the code that returned an empty value because it couldn't fulfil it's task.

I do agree though, it's more of a coding preference.

Reply:
Fair enough, maybe we can continue this discussion some other time.